source/Buoy-Localization-Extensions/String.extension.st

```diff
@@ -0,0 +1,55 @@
+Extension { #name : 'String' }
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> asLanguageRange [
+
+  ^ LanguageRange fromString: self
+]
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> asLanguageTag [
+
+	^ LanguageTag fromString: self
+]
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> escaped [
+
+  ^ self species new: self size streamContents: [ :result |
+        | stream |
+        stream := self readStream.
+        [ stream atEnd ] whileFalse: [ StringEscapingRule escape: stream next on: result ]
+      ]
+]
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> localized [
+
+  ^ self localizedWithAll: #(  )
+]
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> localizedWithAll: collection [
+
+  ^ NaturalLanguageTranslator current localize: self withAll: collection to: CurrentLocale value
+]
+
+{ #category : '*Buoy-Localization-Extensions' }
+String >> unescaped [
+
```

Comment:
Create a separate class EscapingAlgorithm and put the code there, to prevent reimplementing this twice